packages/orm/src/EntityManager.ts

```diff
@@ -435,6 +435,36 @@ export class EntityManager {
     return entityOrList as any;
   }
 
+  /**
+   * Executes `fn` with a transaction, and automatically calls `flush`/`commit` at the end.
+   *
+   * This ensures both any `.find` as well as `.flush` operations happen within the same
+   * transaction, which is useful for enforcing cross-table/application-level invariants that
+   * cannot be enforced with database-level constraints.
+   */
+  public async transaction<T>(fn: () => Promise<T>): Promise<T> {
```

Comment:
I'm not sure you'd want every `em.transaction` to always have `isolation level serializable`.  Perhaps expose it as an optional argument that defaults to serializeable?

Reply:
Yep, agreed...will probably do as a follow up.